yapapi/rest/activity.py

```diff
@@ -167,7 +167,7 @@ async def __aiter__(self) -> AsyncIterator[events.CommandEventContext]:
                 raise BatchTimeoutError()
             try:
                 results: List[yaa.ExeScriptCommandResult] = await self._api.get_exec_batch_results(
-                    self._activity_id, self._batch_id, timeout=timeout
+                    self._activity_id, self._batch_id, _request_timeout=timeout
                 )
             except ApiException as err:
```

Comment:
We should also catch `asyncio.TimeoutError` here and raise `BatchTimeoutError` instead.

Reply:
I would also shorten the amount of sleep in case no new results are available, at the end of this method. It's now 10 seconds:
 https://github.com/golemfactory/yapapi/blob/157e69ed17cede2da6b2490ede042d5490266450/yapapi/rest/activity.py#L197-L198

Reply:
Yes, it's three seconds in `yajsapi`.

Reply:
> In addition to `ApiException` we should also catch `asyncio.TimeoutError` here and raise `BatchTimeoutError` instead.

To avoid (tcp) connection closed errors, I limited maximum timeout to 5 seconds and added `continue` when timeouts happen.